developer_manual/scripts/read-config.php

```diff
@@ -47,26 +46,27 @@ public function __construct($config = '')
 	}
 
 	/**
-	 * Find the first app directory path that is set as
-	 * being writable and is physically writable in the filesystem
+	 * Find a writable app directory path that is either defined by key 'apps_paths'
+	 * or use the default owncloud_root/apps path if the key is not set 
 	 *
 	 * @return string
 	 * @throws \Exception
 	 */
-	function findPath()
-	{
+	function findPath($ocAppsPath)	{
+
+		// default path = /apps
 		if (!array_key_exists('apps_paths', $this->config)) {
-			throw new \Exception('Configuration is missing the apps_path key');
+			$this->output = $ocAppsPath;
+			return $this->output;
```

Comment:
Another thing I overlooked earlier. At this point, correct me if I'm wrong, but as an ownCloud config's passed to the class constructor, it seems redundant to pass a path to the `findPath` method. It seems like it's being passed as a "just in case". Tell me if I'm wrong though.

Reply:
> "just in case"

Not this is not the case.
The constructor delivers the content of config.php
In case there is no apps_path key present, you need the path to owncloud_root because you can identify only then the full path of the /apps directory.